commit 58a744c18bfcb180799843fda78ff78e33c0f16f
Author: bertin <macportsuser@macports.org>
Date:   Sun May 25 22:22:22 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-sqlext.diff

diff --git a/src/cregistry/sqlext.c b/src/cregistry/sqlext.c
index 4c24ea62a..39c359aaf 100644
--- a/src/cregistry/sqlext.c
+++ b/src/cregistry/sqlext.c
@@ -4,6 +4,7 @@
 
 #include "vercomp.h"
 
+#include <stdio.h>
 #include <string.h>
 #include <sqlite3ext.h>
 SQLITE_EXTENSION_INIT1
@@ -19,16 +20,23 @@ SQLITE_EXTENSION_INIT1
 int sqlite3_extension_init(
     sqlite3 *db,          /* The database connection */
     char **pzErrMsg UNUSED,      /* Write error messages here */
-#if HAVE_SQLITE3EXT_H
+#ifdef sqlite3_create_collation
     const sqlite3_api_routines *pApi  /* API methods */
 #else
+#warning "macports.sqlext won't work"
     const void *pApi
 #endif
 ) {
-#if HAVE_SQLITE3EXT_H
+    int rc = SQLITE_OK;
+#ifdef sqlite3_create_collation
     SQLITE_EXTENSION_INIT2(pApi)
 
-    sqlite3_create_collation(db, "VERSION", SQLITE_UTF8, NULL, sql_version);
+    fprintf(stderr, "creating collation sequence VERSION to sql_version() for database \"%s\"\n",
+        sqlite3_db_filename(db, 0));
+    rc = sqlite3_create_collation(db, "VERSION", SQLITE_UTF8, NULL, sql_version);
+    if (rc != SQLITE_OK) {
+	    fprintf(stderr, "Error: %s\n", sqlite3_errmsg(db));
+    }
 #endif
-    return 0;
+    return rc;
 }
